src/IonParserBinaryRaw.ts

```diff
@@ -231,6 +231,52 @@ export class ParserBinaryRaw {
         return ParserBinaryRaw._readUnsignedIntFrom(this._in, this._len);
     }
 
+    /**
+     * Reads an unsigned integer that is small enough to be stored in a Number without losing precision.
+     * This is valuable for use cases like Symbol IDs, which are unlikely to grow beyond a few bytes.
+     *
+     * For simplicity, the current implementation supports reading integers up to 6 bytes (i.e. (2^48) - 1)
+     * rather than Number's maximum safe integer value, 2^53 - 1.     *
+     *
+     * @throws Error if the unsigned int was too large to store in a Number.
+     * @hidden
+     */
+    static _readSafeUnsignedIntFrom(input: BinarySpan, numberOfBytes: number) : number {
+        let value = 0;
+        let bytesRead = 0;
+        let bytesAvailable = input.getRemaining();
+        let byte;
+        if (numberOfBytes < 1) {
+            return 0;
+        } else if (numberOfBytes > 6) {
+            throw new Error(
+                `Attempted to read a ${numberOfBytes}-byte unsigned integer,`
+                + ` which is too large for a to be stored in a number without losing precision.`
+            );
+        }
+
+        if (bytesAvailable < numberOfBytes) {
+            throw new Error(
+                `Attempted to read a ${numberOfBytes}-byte unsigned integer,`
+                + ` but only ${bytesAvailable} bytes were available.`
+            );
+        }
+
+        while (bytesRead < numberOfBytes) {
+            byte = input.next();
+            bytesRead++;
+            // Avoid using bitshifting to preserve Number's precision beyond 31 bits.
+            value *= 256;
+            value = value + byte;
```

Comment:
It *might* be worth using integer machinery for the vast amount of symbols versus floating point multiply for the case where we grow beyond signed 31-bits.  It does add complexity and I doubt we have benchmarks here, so maybe even a TODO to consider it might be all that is warranted.

Reply:
Agreed. Wes pointed me to [this performance comparison](https://jsperf.com/multiplication-int-vs-bit-shifting-left/4).

I'll add a TODO.